src/envoy/http/path_rewrite/config_parser_impl.cc

```diff
@@ -0,0 +1,139 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/envoy/http/path_rewrite/config_parser_impl.h"
+
+#include "absl/strings/str_cat.h"
+#include "common/common/empty_string.h"
+#include "src/api_proxy/path_matcher/variable_binding_utils.h"
+
+namespace espv2 {
+namespace envoy {
+namespace http_filters {
+namespace path_rewrite {
+namespace {
+
+// Use fixed HTTP method for path_matcher
+constexpr const char kHttpMethod[] = "GET";
+
+}  // namespace
+
+ConfigParserImpl::ConfigParserImpl(
+    const ::espv2::api::envoy::v9::http::path_rewrite::PerRouteFilterConfig&
+        config)
+    : config_(config) {
+  if (config_.has_constant_path()) {
+    const auto& path_cfg = config_.constant_path();
+    if (!path_cfg.url_template().empty()) {
+      ENVOY_LOG(debug, "Building path_matcher for url_template: {}",
+                path_cfg.url_template());
+
+      ::espv2::api_proxy::path_matcher::PathMatcherBuilder<
+          const ::espv2::api::envoy::v9::http::path_rewrite::
+              PerRouteFilterConfig*>
+          pmb;
+      pmb.Register(kHttpMethod, path_cfg.url_template(), Envoy::EMPTY_STRING,
+                   &config_);
+      path_matcher_ = pmb.Build();
+    }
+
+    // If the last char of the path is "/", remove it, unless it is just root
```

Comment:
Either place is fine.  Since it is easy to add at filter level, it makes it easier to config it.  I will feel more secure if filter level code handle it correctly.